pyas2lib/as2.py

```diff
@@ -372,10 +377,13 @@ def build(
                 "Encryption of messages is enabled but encrypt key is not set for the receiver."
             )
 
-        # Generate message id using UUID 1 as it uses both hostname and time
-        self.message_id = (
-            email_utils.make_msgid(domain=self.sender.domain).lstrip("<").rstrip(">")
-        )
+        if message_id:
+            self.message_id = message_id  
```

Comment:
as per the rfc the message id needs to follow a particular format, if we are allowing users to do it then it needs to be validated

```
   5.3.3.  Message-Id and Original-Message-Id

   Message-Id and Original-Message-Id is formatted as defined in RFC
   2822 [9]:

          "<" id-left "@" id-right ">"        (RFC 2822, 3.6.4)

   Message-Id length is a maximum of 998 characters.  For maximum
   backward compatibility, Message-Id length SHOULD be 255 characters or
   less.  Message-Id SHOULD be globally unique, and id-right SHOULD be
   something unique to the sending host environment (e.g., a host name).

   When sending a message, always include the angle brackets.  Angle
   brackets are not part of the Message-Id value.  For maximum backward
   compatibility, when receiving a message, do not check for angle
   brackets.  When creating the Original-Message-Id header in an MDN,
   always use the exact syntax as received on the original message;
   don't strip or add angle brackets.
```

Reply:
I've modified this a bit based on your feedback.  Let me know if that's closer to what you want.  Thanks for your guidance!

Reply:
We have partners we're integrating with that are dictating the format of the message ids that we send.  In my immediate case, the partner is requiring a specific format to be sent ex: `{specific_customer_identifier}.{unique_id_part}@example.org`

We're seeing two very large companies recently asking us to do this for all of our customer's connections.  By allowing the message id to be customized we can adhere to our partners' requirements.  Ideally this shouldn't be something that they would require at all (it is a weird requirement), but we've routinely needed to be flexible as they have most of the control over how these things work. 

Hope that makes sense, and thank you so much for such a useful library!

Reply:
Makes sense